src/scenario_tests.rs

```diff
@@ -871,11 +885,77 @@ mod dst_tests {
             "Merge if we detect our section needs merging on CheckElder",
             &initial_state,
             &[
-                TestEvent::SetMergeNeeded(true).to_event(),
+                TestEvent::SetMergeNeeded(MergeInfo(OTHER_SECTION_INFO)).to_event(),
+                ParsecVote::CheckElder.to_event(),
+            ],
+            &AssertState {
+                action_our_events: vec![Rpc::Merge(OTHER_SECTION_INFO).to_event()],
+            },
+        );
+    }
+
+    #[test]
+    fn parsec_merge_sibling() {
+        let initial_state = arrange_initial_state(
+            &initial_state_old_elders(),
+            &[
+                TestEvent::SetMergeNeeded(MergeInfo(OTHER_SECTION_INFO)).to_event(),
+                ParsecVote::CheckElder.to_event(),
+            ],
+        );
+        run_test(
+            "Decide to merge, and then later store merge infos in ProcessMerge",
+            &initial_state,
+            &[ParsecVote::NeighbourMerge(MergeInfo(OTHER_SECTION_INFO)).to_event()],
```

Comment:
Yes, that one should be different from our section info that we are sending in the RPC.